osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -431,6 +434,24 @@ public async Task StartMatch()
             }
         }
 
+        public async Task AbortGameplay()
+        {
+            using (var userUsage = await GetOrCreateLocalUserState())
+            using (var roomUsage = await getLocalUserRoom(userUsage.Item))
+            {
+                var room = roomUsage.Item;
+                if (room == null)
+                    throw new InvalidOperationException("Attempted to operate on a null room");
+
+                var user = room.Users.FirstOrDefault(u => u.UserID == CurrentContextUserId);
+                if (user == null)
+                    throw new InvalidOperationException("Local user was not found in the expected room");
+
+                await changeAndBroadcastUserState(room, user, MultiplayerUserState.Idle);
```

Comment:
Is this intentionally not doing any pre-checks? ie. the client could call this from a `Ready` or `Results` state and it would still set them to idle. Not sure that's the behaviour we're looking for.

It also means the following like `updateRoomStateIfRequired` is running even if there is no state change. Looks like that won't be an issue currently, but it's different to how we've done it until now so not 100%.

Reply:
I didn't think it was that big of a deal, but fair enough. I've added validation.